File: first-unique_python/firstUnique/firstUnique.py


def first_unique(str):
    for pos in range(len(str)):
        if is_unique(pos, str):
            return str[pos]
    return None


def is_unique(pos, str):
    gen = [i for i in range(len(str)) if i != pos]
    for p in gen:
        if str[p] == str[pos]:
            return False
    return True

File: first-unique_scala/src/main/scala/com/aguiot/firstUnique/FirstUnique.scala
package com.aguiot.firstUnique

object FirstUnique {
  def firstUnique(str: String): Option[Char] = {
    str.indices.find(isUnique(_, str)) match {
      case Some(i) => Some(str(i))
      case None => None
    }
  }

  def isUnique(pos: Int, str: String): Boolean = {
    for {
      i <- 0 until str.length
      if i != pos
    } {
      if (str(i) == str(pos)) return false
    }
    true
  }
}
